media-sound/guayadeque/guayadeque-0.4.3.ebuild

```diff
@@ -0,0 +1,106 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=6
+
+WX_GTK_VER=3.0
+
+inherit cmake-utils wxwidgets
+
+DESCRIPTION="Music management program designed for all music enthusiasts"
+HOMEPAGE="http://guayadeque.org/"
+SRC_URI="https://github.com/anonbeat/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="ayatana ipod +minimal"
+
+# No test available, Making src_test fail
+RESTRICT="test"
+
+GST_VERSION=1.0
+GST_DEPS="
+	dev-perl/GStreamer
+	media-libs/gnonlin:${GST_VERSION}
+	media-plugins/gst-plugins-libav:${GST_VERSION}
+	media-plugins/gst-plugins-libnice:${GST_VERSION}
+	media-plugins/gst-plugins-pulse:${GST_VERSION}
+	media-plugins/gst-plugins-soup:${GST_VERSION}
+	media-libs/gst-plugins-bad:${GST_VERSION}
+	media-libs/gst-plugins-base:${GST_VERSION}
+	media-libs/gst-plugins-good:${GST_VERSION}
+	media-libs/gst-plugins-ugly:${GST_VERSION}
+"
+
+RDEPEND="
+	dev-db/sqlite:3
+	>=dev-db/wxsqlite3-3.2.1
+	dev-libs/glib:2
+	media-libs/flac
+	media-libs/gstreamer:${GST_VERSION}[introspection]
+	>=media-libs/taglib-1.6.4
+	net-misc/curl
+	sys-apps/dbus
+	x11-libs/wxGTK:${WX_GTK_VER}[X]
+	ayatana? ( >=dev-libs/libindicate-0.7 )
+	ipod? ( media-libs/libgpod )
+	!minimal? ( ${GST_DEPS} )"
+DEPEND="${RDEPEND}
+	app-arch/unzip
+	sys-devel/gettext
+	virtual/pkgconfig"
+
+PATCHES=(
+	"${FILESDIR}"/${PN}-0.4.3-fix-build-system.patch
+	"${FILESDIR}"/${PN}-0.4.3-fix-ifdef.patch
+	"${FILESDIR}"/${PN}-0.4.3-desktop.patch
+)
+
+# echo $(cat po/CMakeLists.txt | grep ADD_SUBDIRECTORY | sed 's#ADD_SUBDIRECTORY( \(\w\+\) )#\1#')
+LANGS=( bg ca_ES cs de el es fr hr hu is it ja nb nl pl pt pt_BR ru sk sr sr@latin sv th tr uk )
+for l in "${LANGS[@]}"; do
+	IUSE+=" linguas_${l}"
+done
+unset l
+
+pkg_setup() {
+	setup-wxwidgets
+}
+
+src_prepare() {
+	cmake-utils_src_prepare
+
+	mv changelog ChangeLog || die
+
+	# remove bundled libs
+	rm -rf src/wx/wxsql* src/wxsqlite3 || die
+
+	# comment out unused languages
+	cd po || die
+	local l
+	for l in "${LANGS[@]}"; do
+		! use linguas_${l} && cmake_comment_add_subdirectory ${l}
+	done
+}
+
+src_configure() {
+	local mycmakeargs=(
+		-DENABLE_IPOD=$(usex ipod)
+		-DENABLE_LIBINDICATE=$(usex ayatana)
+	)
+	cmake-utils_src_configure
+}
+
+pkg_postinst() {
+	if use minimal; then
+		elog "If you are missing functionalities consider setting USE=-minimal"
+		elog "or install any of the following packages:"
+
```

Comment:
The problem with this instruction is that while you can add plugin packages, some are really important to have. I think if gstreamer is already a hard dependency, outside of GST_DEPS variables, then at least gst-plugins-base and gst-plugins-good should be moved there as a hard dependency as well (*-good is more debatable, but good to at least have autoaudiosink and such, and everyone's going to have gst-plugins-good anyways if they want to do anything useful with gstreamer anyways)